app/pages/OrganizationEditPage.tsx

```diff
@@ -1329,17 +1329,17 @@ class OrganizationEditPage extends React.Component<Props, State> {
 
   handleDeactivation(type: "resource" | "service", id: number): void {
     const { history } = this.props;
-    let confirmMessage: string | null = null;
+    let confirmMessage: string | undefined;
     let path: string | null = null;
```

Comment:
Looking at this code more carefully, we can actually avoid having a `| null` or `| undefined` in the types at all if we just throw an exception in the `if` and `else if` chains. If you add this after the `else if` block

```typescript
    } else {
      throw new Error(`Unexpected type: ${type}`);
    }
```

then both `confirmMessage` and `path` can have a type of just `string` and we don't need to set an initial value here. The reason this works is because we're proving to TypeScript that either the `type === "resource"` or the `type === "service"` branch has to be taken, and therefore both `confirmMessage` and `path` are guaranteed to be given a real value. This should also allow you to remove the `as` assertions on them later in this function.

If it's not too much trouble, could you make this change instead of the change you had here?

Reply:
Good thinking. Done!